### ISSUE TITLE
Scan error no readable memory found

### ISSUE THREAD

**Issue by Kiritoking1** [cheat-engine/cheat-engine#2868]
Whenever I try to scan it shows this error, help.

**Comment by cheat-engine** [comment]
mac or windows?

if mac and the target is rosetta,  go to settings and tick mem_mapped

**Comment by Kiritoking1** [comment]
Wdym target is Rosetta

**Comment by cheat-engine** [comment]
x86 based processes on an arm based mac  run under the rosetta emulator.

**Comment by ntdungqlqb** [comment]
> mac or windows?
> 
> if mac and the target is rosetta, go to settings and tick mem_mapped


where mem_mapped

**Comment by cheat-engine** [comment]
settings->scan settings,  and then near the bottom

**Comment by HitmanForlife** [comment]
I didn't understand what you meant so can u maybe explain it briefly, sorry [cheat-engine]

**Comment by RBMLWizard** [comment]
Fix for "Scan error: No readable memory found"

### Issue
The error occurs when a 32-bit version is incompatible with some applications. 

###Solution
Use the 64-bit executable rather than the 32-bit version.

(Optional)
### Build Solution
Use the 64bit executable.

1. Open Lazarus IDE.
2. Go to `Project` > `Project Options` > `Compiler Options` > `Building`.
3. Select `Build many modes` instead of `Build`.
4. Build your project with the 64-bit target.

This ensures compatibility with 64-bit applications and avoids the error.

**Comment by Guoyutian** [comment]
> settings->scan settings, and then near the bottom

hello,i haved tapped the mem_mapped, but not works

![Image](https://github.com/user-attachments/assets/86daf223-9000-4f7d-b489-87232226fea9)

![Image](https://github.com/user-attachments/assets/be634f3f-0118-40a9-b7d2-52c30537e6dd) macos sonoma and 7.5ce，i have no idea now

**Comment by cheat-engine** [comment]
Are you sure that the memory at that address is supposed to be readable?

**Comment by Guoyutian** [comment]
absolutely,it works on win using 7.6ce, i have tried tons of times on macos sonoma(x86 chip) with my friends, but ,,,,,,

![Image](https://github.com/user-attachments/assets/1c282e36-4673-4db5-958f-868b27bf1390)

it works on win, and the app is weixin

![Image](https://github.com/user-attachments/assets/52509b8e-4e23-4a71-9347-78c72ee13511)on win

<img width="1163" alt="Image" src="https://github.com/user-attachments/assets/eebb3715-65a3-4670-8d5f-502832d286b3" />on macos

**Comment by Guoyutian** [comment]
works on win

![Image](https://github.com/user-attachments/assets/3f843110-ac41-4a35-9e05-30ad255d12ba)

**Comment by Square-funny** [comment]
same issue here, when i selected rosetta game on mac the CE report the same, turn on the mem_mapped did not help at first, but restart the whole process solved the problem.

**Comment by ARC4991** [comment]
I also encountered this issue, and it was resolved after enabling `mem_mapped` and restarting the program.

**Comment by iSavra** [comment]
@cheat-engine Please enable mem_mapped by default on the macOS release, you'll avoid endless confusion and frustration to new users, Cheat Engine was 100% nonfunctional on my case until I finally found this solution!
